lib/xdrgen/generators/go.rb

```diff
@@ -580,8 +815,16 @@ def render_top_matter(out)
             xdrType()
           }
 
+          type xdrDecodable interface {
```

Comment:
❗ Could you add for every type a cast to an unnamed variable? We do that with all the other interfaces in here to make sure all the types satisfy the interfaces. i.e.:

```go
var _ xdrDecodeable = (*#{name})(nil)
```

Reply:
❗ Could you name this interface `decoderFrom`? That's the pattern used for similar named interfaces in the stdlib. e.g. The interface that contains the `ReadFrom` function is [ReaderFrom](https://pkg.go.dev/io#ReaderFrom).

Reply:
Makes sense, I will also add it for `EncodeTo`

Reply:
Will do

Reply:
+1 As a follow up we can remove the xdrType thing I added too, it's superfluous and wasn't the best approach I took. I'm happy to do the clean up for EncodeTo, but if you're going to do it 👍🏻 sgtm.

Reply:
If you could cleanup the encodeto part (adding errors, making it consistent with this etc ...). I would appreciate it.